bot/seasons/evergreen/movie.py

```diff
@@ -0,0 +1,224 @@
+import logging
+import random
+from enum import Enum
+from os import environ
+from typing import List, Tuple
+from urllib.parse import urlencode
+
+from aiohttp import ClientSession
+from discord import Embed
+from discord.ext.commands import Bot, Cog, Context, command
+
+from bot.pagination import ImagePaginator
+
+# Get TMDB API key from .env
+TMDB_API_KEY = environ.get('TMDB_API_KEY')
+
+# Define base URL of TMDB
+BASE_URL = "https://api.themoviedb.org/3/"
+
+# Get logger
+logger = logging.getLogger(__name__)
+
+
+# Genres with TMDB API IDs
+class MovieGenres(Enum):
+    """Genre names and IDs."""
+
+    Action = 28
+    Adventure = 12
+    Animation = 16
+    Comedy = 35
+    Crime = 80
+    Documentary = 99
+    Drama = 18
+    Family = 10751
+    Fantasy = 14
+    History = 36
+    Horror = 27
+    Music = 10402
+    Mystery = 9648
+    Romance = 10749
+    Science = 878
+    Thriller = 53
+    Western = 37
+
+
+async def get_random_movies(client: ClientSession,
+                            count: int,
+                            genre_id: int,
+                            genre_name: str,) \
+        -> Tuple[List[Tuple[str, str]], Embed]:
+    """Get random movies by genre from TMDB."""
+    pages = []
+
+    # Create embed
+    embed = Embed(title=f"Random {genre_name} Movies")
+    embed.set_footer(text='Powered by TMDB (themoviedb.org)')
+
+    # Get random page between 1 and 200
+    page = random.randint(1, 200)
+
+    # Define TMDB request parameters
+    # (API key, exclusions, inclusions, sort)
+    params = {
+        "api_key": TMDB_API_KEY,
+        "language": "en-US",
+        "sort_by": "popularity.desc",
+        "include_adult": "false",
+        "include_video": "false",
+        "page": page,
+        "with_genres": str(genre_id)
+    }
+
+    # Create request URL
+    url = BASE_URL + "discover/movie" + "?" + urlencode(params)
+
+    # Do discover request to TMDB API and fetch information
+    async with client.get(url) as res:
+        try:
+            # Parse response data from JSON to dictionary
+            movie_list_data = await res.json()
+
+            # Loop and fetch movies
+            for i in range(count):
+                # Get movie ID
+                movie_id = movie_list_data["results"][i]["id"]
+
+                # Create movie params
+                movie_params = {
+                    "api_key": TMDB_API_KEY,
+                    "language": "en-US"
+                }
+
+                # Generate URL
+                movie_url = BASE_URL + f"movie/{movie_id}?" + urlencode(
+                    movie_params)
+
+                # Fetch movie
+                async with client.get(movie_url) as m_res:
+                    # Parse JSON to dict
+                    movie_data = await m_res.json()
```

Comment:
The with block can be exited and the connection released after getting the JSON.